### ISSUE TITLE
Stackstrings loaded into global variables

### ISSUE THREAD

**Issue by mr-tz** [fireeye/flare-floss#266]
See function in 53bcda3e3fa629163328911e6ca13445 at VA 0x100013B0.

**Comment by williballenthin** [comment]
see https://github.com/fireeye/flare-floss/tree/fix/get_stackstrings_on_move for potential solution

performance changes:
```
Finished execution after 2.001157 seconds | Finished execution after 0.889498 seconds
Finished execution after 1.535905 seconds | Finished execution after 0.534664 seconds
Finished execution after 7.279704 seconds | Finished execution after 3.540252 seconds
Finished execution after 1.210376 seconds | Finished execution after 0.876422 seconds
Finished execution after 1.059937 seconds | Finished execution after 1.560157 seconds
Finished execution after 48.164911 seconds | Finished execution after 32.614481 seconds
Finished execution after 13.890418 seconds | Finished execution after 9.625995 seconds
Finished execution after 12.354010 seconds | Finished execution after 16.892529 seconds
Finished execution after 30.181431 seconds | Finished execution after 31.138448 seconds
Finished execution after 29.413580 seconds | Finished execution after 26.919828 seconds
Finished execution after 9.784146 seconds | Finished execution after 10.114317 seconds
Finished execution after 2.675725 seconds | Finished execution after 1.191114 seconds
Finished execution after 3.308836 seconds | Finished execution after 1.253435 seconds
Finished execution after 1.394621 seconds | Finished execution after 0.298292 seconds
Finished execution after 1.644238 seconds | Finished execution after 0.301401 seconds
Finished execution after 1.421414 seconds | Finished execution after 1.160257 seconds
Finished execution after 15.485367 seconds | Finished execution after 18.499221 seconds
Finished execution after 8.429687 seconds | Finished execution after 9.152752 seconds
Finished execution after 5.956419 seconds | Finished execution after 9.267610 seconds
Finished execution after 0.107332 seconds | Finished execution after 0.120155 seconds
Finished execution after 0.310461 seconds | Finished execution after 0.401672 seconds
```

**Comment by williballenthin** [comment]
performance decrease by a factor of 2x is not very good. we should research heuristics to detect when to apply this technique.

**Comment by williballenthin** [comment]
potential heuristics:

detecting a sequence of four "move single byte constant into contiguous buffer" seems like it should be doable. then trigger a collection at the end of the current basic block?

**Comment by mr-tz** [comment]
Getting much worse https://github.com/fireeye/flare-floss/commit/9f2ceb2a24066f58131a9628c95ac80467544331 :)
```
Finished execution after 1.763606 seconds		      |	Finished execution after 1.050786 seconds
Finished execution after 1.706242 seconds		      |	Finished execution after 0.629037 seconds
Finished execution after 12.561846 seconds		      |	Finished execution after 1.522367 seconds
Finished execution after 2.636911 seconds		      |	Finished execution after 0.990657 seconds
Finished execution after 2.355830 seconds		      |	Finished execution after 1.464837 seconds
Finished execution after 108.653779 seconds		      |	Finished execution after 32.578955 seconds
Finished execution after 30.139329 seconds		      |	Finished execution after 5.361783 seconds
Finished execution after 31.426122 seconds		      |	Finished execution after 4.235152 seconds
Finished execution after 58.432218 seconds		      |	Finished execution after 28.328247 seconds
Finished execution after 60.083270 seconds		      |	Finished execution after 23.393540 seconds
Finished execution after 23.200419 seconds		      |	Finished execution after 8.643929 seconds
Finished execution after 5.616619 seconds		      |	Finished execution after 3.125362 seconds
Finished execution after 12.671287 seconds		      |	Finished execution after 4.440304 seconds
Finished execution after 2.808645 seconds		      |	Finished execution after 1.345349 seconds
Finished execution after 2.712589 seconds		      |	Finished execution after 1.272226 seconds
Finished execution after 10.651504 seconds		      |	Finished execution after 0.928227 seconds
Finished execution after 32.426640 seconds		      |	Finished execution after 11.923773 seconds
Finished execution after 18.479772 seconds		      |	Finished execution after 6.459448 seconds
Finished execution after 17.081412 seconds		      |	Finished execution after 9.025540 seconds
Finished execution after 0.674579 seconds		      |	Finished execution after 0.120631 seconds
Finished execution after 1.554304 seconds		      |	Finished execution after 0.892557 seconds
```

**Comment by mr-tz** [comment]
Another approach could be to use vivisect's writelog.

**Comment by mr-tz** [comment]
For 4ae6fc176e4ce97b360246328012c97106e3e02e
Don't ask me why it's sometimes faster with the update. I suspect it might be failing although the extracted stack strings are almost identical...
```
Before                                                                              After
FLOSS extracted 68 stackstrings				      |	FLOSS extracted 70 stackstrings
Finished execution after 21.395251 seconds		      |	Finished execution after 11.779689 seconds
FLOSS extracted 68 stackstrings				      |	FLOSS extracted 70 stackstrings
Finished execution after 12.086730 seconds		      |	Finished execution after 5.766737 seconds
FLOSS extracted 33 stackstrings					FLOSS extracted 33 stackstrings
Finished execution after 2.909764 seconds		      |	Finished execution after 2.250041 seconds
FLOSS extracted 33 stackstrings					FLOSS extracted 33 stackstrings
Finished execution after 7.052175 seconds		      |	Finished execution after 2.185107 seconds
FLOSS extracted 34 stackstrings					FLOSS extracted 34 stackstrings
Finished execution after 2.014953 seconds		      |	Finished execution after 2.196353 seconds
FLOSS extracted 7 stackstrings					FLOSS extracted 7 stackstrings
Finished execution after 0.541726 seconds		      |	Finished execution after 0.534595 seconds
FLOSS extracted 58 stackstrings					FLOSS extracted 58 stackstrings
Finished execution after 2.123646 seconds		      |	Finished execution after 2.852025 seconds
FLOSS extracted 33 stackstrings					FLOSS extracted 33 stackstrings
Finished execution after 1.869085 seconds		      |	Finished execution after 2.126633 seconds
FLOSS extracted 6 stackstrings					FLOSS extracted 6 stackstrings
Finished execution after 0.276731 seconds		      |	Finished execution after 0.273309 seconds
FLOSS extracted 4 stackstrings					FLOSS extracted 4 stackstrings
Finished execution after 0.079342 seconds		      |	Finished execution after 0.093086 seconds
FLOSS extracted 0 stackstrings					FLOSS extracted 0 stackstrings
Finished execution after 0.321545 seconds		      |	Finished execution after 0.298798 seconds
FLOSS extracted 5 stackstrings					FLOSS extracted 5 stackstrings
Finished execution after 0.964567 seconds		      |	Finished execution after 1.415666 seconds
FLOSS extracted 2 stackstrings					FLOSS extracted 2 stackstrings
Finished execution after 0.125332 seconds		      |	Finished execution after 0.119400 seconds
FLOSS extracted 0 stackstrings					FLOSS extracted 0 stackstrings
Finished execution after 0.310682 seconds		      |	Finished execution after 0.376916 seconds
FLOSS extracted 12 stackstrings					FLOSS extracted 12 stackstrings
Finished execution after 6.055719 seconds		      |	Finished execution after 1.704970 seconds
FLOSS extracted 0 stackstrings					FLOSS extracted 0 stackstrings
Finished execution after 0.627058 seconds		      |	Finished execution after 0.262685 seconds
FLOSS extracted 0 stackstrings					FLOSS extracted 0 stackstrings
Finished execution after 0.199009 seconds		      |	Finished execution after 0.110610 seconds
FLOSS extracted 4 stackstrings					FLOSS extracted 4 stackstrings
Finished execution after 0.300824 seconds		      |	Finished execution after 0.172792 seconds
FLOSS extracted 30 stackstrings					FLOSS extracted 30 stackstrings
Finished execution after 9.836497 seconds		      |	Finished execution after 4.135498 seconds
FLOSS extracted 26 stackstrings					FLOSS extracted 26 stackstrings
Finished execution after 6.628611 seconds		      |	Finished execution after 2.229550 seconds
FLOSS extracted 12 stackstrings					FLOSS extracted 12 stackstrings
Finished execution after 2.260597 seconds		      |	Finished execution after 3.112243 seconds
FLOSS extracted 23 stackstrings					FLOSS extracted 23 stackstrings
Finished execution after 1.667418 seconds		      |	Finished execution after 2.097401 seconds
```

**Comment by williballenthin** [comment]
follow #269 for PR implementing these changes.

**Comment by mr-tz** [comment]
Closed via #269.
